### ISSUE TITLE
data grid premium doesn't support ctrl + z to undo changes

### ISSUE THREAD

**Issue by simonesanna1983** [mui/mui-x#9339]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Hi everyone,

I was trying the premium demo datagrid. I have noticed the clipboard features is working very well.
I have also tried to press ctrl + A and it works fine.
Unfortunately the ctrl + Z feature to undo changes doesn't work, it's probably something not implemented yet.

I would like to ask if it is something to consider as some reference products include this feature. 
Could it be something to put into the roadmap?

Thanks in advance.

**Comment by cherniavskii** [comment]
As an initial implementation, we want to narrow it down to the clipboard paste only.
We want to support both `Ctrl(Cmd)+Z` (Undo) and `Ctrl+(Cmd)+Shift+Z` (and potentially `Ctrl(Cmd)+Y`) shortcuts in the Premium plan.

**Comment by md-abdul-halim-rafi** [comment]
Any latest update on this feature?

**Comment by jjram19** [comment]
any update here?

**Comment by simonesanna1983** [comment]
are you guys still there?

**Comment by tmackarel-r7** [comment]
This feature would also be useful for my company's project, specifically with copy/paste.

**Comment by cherniavskii** [comment]
We need to do some benchmarking first, then define what a single Operation is, and ensure that each Undo operation is also persisted.

**Comment by anthonysette** [comment]
Agree, this would be very valuable out of the box. Great premium feature
